tests/test_integration.py

```diff
@@ -445,225 +496,161 @@ def _get_vm():
-        cur_hv = self.vm.server_obj['xen_host']
-        new_hv = query(
-            servertype='hypervisor',
-            environment='testing',
-            vlan_networks=self.vm.server_obj['route_network'],
-            hostname=Not(cur_hv)
-        ).get()['hostname']
-
-        migratevm(self.vm.server_obj['hostname'])
-
-        self.vm.reload()
-        self._check_vm(Hypervisor(new_hv), self.vm)
-        self._check_absent(Hypervisor(cur_hv), self.vm)
-
-        # And back again otherwise the tearDown will fail WTF!
-        cur_hv = self.vm.server_obj['xen_host']
-        new_hv = query(
-            servertype='hypervisor',
-            environment='testing',
-            vlan_networks=self.vm.server_obj['route_network'],
-            hostname=Not(cur_hv)
-        ).get()['hostname']
-
-        migratevm(self.vm.server_obj['hostname'])
-
-        self.vm.reload()
-        self._check_vm(Hypervisor(new_hv), self.vm)
-        self._check_absent(Hypervisor(cur_hv), self.vm)
+        # auto find means no target HV is specified
+        migratevm(self.vm_obj['hostname'])
+        self.check_vm_present()
 
     def test_offline_migration(self):
-        migratevm(self.vm.server_obj['hostname'], HV2, offline=True)
-        self.vm.reload()
-        self._check_vm(self.hv2, self.vm)
-        self._check_absent(self.hv1, self.vm)
-
-        # And back again
-        migratevm(self.vm.server_obj['hostname'], HV1, offline=True)
-        self.vm.reload()
-        self._check_vm(self.hv1, self.vm)
-        self._check_absent(self.hv2, self.vm)
+        migratevm(self.vm_obj['hostname'], self.new_hv_name, offline=True)
+        self.check_vm_present()
 
     def test_reject_out_of_sync_serveradmin(self):
-        self.vm.server_obj['disk_size_gib'] += 1
-        self.vm.server_obj.commit()
+        self.vm_obj['disk_size_gib'] += 1
+        self.vm_obj.commit()
 
         with self.assertRaises(InconsistentAttributeError):
-            migratevm(self.vm.server_obj['hostname'], HV2)
+            migratevm(self.vm_obj['hostname'], self.new_hv_name)
 
     def test_reject_online_with_new_ip(self):
         with self.assertRaises(IGVMError):
-            migratevm(self.vm.server_obj['hostname'], HV2, newip=IP2)
+            # Fake IP address is fine, this is a failing test.
+            migratevm(
+                self.vm_obj['hostname'], self.new_hv_name,
+                newip='1.2.3.4',
+            )
 
     def test_reject_new_ip_without_puppet(self):
         with self.assertRaises(IGVMError):
+            # Fake IP address is fine, this is a failing test.
             migratevm(
-                self.vm.server_obj['hostname'], HV2, offline=True, newip=IP2
+                self.vm_obj['hostname'], self.new_hv_name, offline=True,
+                newip='1.2.3.4',
             )
 
-    @unittest.skip("depends on automatic regeneration of ig.local domain")
+    @unittest.skip("Broken until we have new Serveradmin API deployed")
```

Comment:
Let's try enabling this now.

Reply:
Let's not enable it for now. I'm afraid we might loose hours trying to fix it. And since the test was always disabled, we don't make things worse.